lib/aws-signer.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Note: much of this code is from https://github.com/carsales/aws-request-signer/blob/c835ca69e74a5b5e7d6eee7aaab8bbd7fbcdff4d/src/signer.js
+ * licensed under https://github.com/carsales/aws-request-signer/blob/c835ca69e74a5b5e7d6eee7aaab8bbd7fbcdff4d/LICENSE
+ * with many thanks to John Jones :+1:
+ */
+
```

Comment:
I appreciate that this file is mainly a copy from elsewhere but could we please update it to make the code style in the rest of the repo. This is partly my bad for not yet adding proper liniting